src/Kernel-Tests.st

```diff
@@ -1167,6 +1178,44 @@ testLastN
 	self should: [ self collection last: 33 ] raise: Error
 !
 
+testRemoveAllFound
+
+	| original part |
+	
+	original := #( 1 2 3 4 5 6 7 8 9 ).
+	part := #( 2 4 6 ).
+	
+	original removeAllFoundIn: part.
+
+	self shouldnt: [ self assert: (part allSatisfy: [ :e | (original includes: e) not ]) ] raise: Error.
+	self shouldnt: [ original removeAllFoundIn: #( 11 12 ) ] raise: Error
+!
+
+testRemoveAllSuch
+
+	| original part |
+	
+	original := #( 1 2 3 4 5 6 7 8 9 ).
+	part := #( 2 4 6 ).
+	
+	original removeAllSuchThat: [ :e | part includes: e ].
+
+	self shouldnt: [ self assert: (part allSatisfy: [ :e | (original includes: e) not ]) ] raise: Error.
+!
+
+testRemoveAllUsingSome
+
+	| original part |
+	
+	original := #( 1 2 3 4 5 6 7 8 9 ).
+	part := #( 2 4 6 ).
+	
+	original removeAll: part.
+
+	self shouldnt: [ self assert: (part allSatisfy: [ :e | (original includes: e) not ]) ] raise: Error.
+	self should: [ original removeAll: #( 11 12 ) ] raise: Error
+!
+
```

Comment:
Those are suboptimal tests. There are many types of collections. There are tests that are written more generically and are inherited, thus testing many types of collections. Look at how other tests are done for Collection hierarchy.